src/components/pages/common/image-viewer.tsx

```diff
@@ -0,0 +1,193 @@
+import { ChevronLeftIcon, ChevronRightIcon, CloseIcon } from "@chakra-ui/icons";
+import { Box, Button, Icon, Image } from "@chakra-ui/react";
+import styled from "@emotion/styled";
+import DeleteIcon from "@icons/delete";
+import { axDeleteResource } from "@services/media-gallery.service";
+import { axGetLicenseList } from "@services/resources.service";
+import { adminOrAuthor } from "@utils/auth";
+import { getResourceRAW } from "@utils/media";
+import notification, { NotificationType } from "@utils/notification";
+import React, { useEffect, useState } from "react";
+
+import CarouselResourceInfo from "./resource-info";
+
+const WatermarkBottomRight = styled.span`
+  position: absolute;
+  bottom: 10px;
+  right: 10px;
+  font-size: 40px;
+  color: white;
+  opacity: 0.5;
+`;
+const WatermarkBottomLeft = styled.span`
+  position: absolute;
+  bottom: 10px;
+  left: 10px;
+  font-size: 40px;
+  color: white;
+  opacity: 0.5;
+`;
+
+const ImageViewer = ({ resourceData, initialIndex, onClose, loadNextPage }) => {
+  const [currentIndex, setCurrentIndex] = useState(initialIndex);
+
+  const [licensesList, setLicenseList] = useState([]);
+
+  useEffect(() => {
+    const fetchLicenses = async () => {
+      try {
+        const { data: licenses } = await axGetLicenseList();
+        setLicenseList(licenses);
+      } catch (error) {
+        console.error("Error fetching licenses:", error);
+      }
+    };
+
+    fetchLicenses();
+  }, []);
+
+  const imageUrls = resourceData.l.map((o) => getResourceRAW("RESOURCE", o.resource?.fileName));
+
+  const caption = resourceData.l.map((o) => o.resource?.description);
+
+  const resourceWaterMark = resourceData.l.map((o) => o.resource?.contributor || o.userIbp?.name);
+
+  const currentResource = resourceData.l[currentIndex];
+
+  const hasAccess = adminOrAuthor(currentResource?.resource.uploaderId);
+
+  const handleNext = async () => {
+    if (currentIndex === resourceData.l.length - 1) {
```

Comment:
can we use ternary operator here instead  of if-else?

Reply:
for async and await we are following if else only and also it is not recommend to use ternary because these async functions will return a promise but ternary expects a value to be assigned .